src/System.Transactions/tests/CloneTxTests.cs

```diff
@@ -115,9 +124,34 @@ public void Run(CloneType cloneType, IsolationLevel isoLevel, bool forcePromote,
             {
                 tx.Commit();
             }
-            catch (TransactionAbortedException)
+            catch (TransactionAbortedException ex)
             {
                 Assert.Equal(expectedStatus, TransactionStatus.Aborted);
+                switch (cloneType)
+                {
+                    case CloneType.Normal:
+                        {
+                            // We shouldn't be getting TransactionAbortedException for "normal" clones,
+                            // so we have these two Asserts to possibly help determine what went wrong.
+                            Assert.Null(ex.InnerException);
+                            Assert.Equal(ex.Message, "There shouldn't be any exception with this Message property");
+                            break;
+                        }
+                    case CloneType.BlockingDependent:
+                        {
+                            Assert.Equal(ex.InnerException.GetType(), typeof(TimeoutException));
```

Comment:
Thanks. Making that change here and a few other places.

---
In reply to: [82718724](https://github.com/dotnet/corefx/pull/12533#discussion_r82718724) [](ancestors = 82718724)